scripts/cdist

```diff
@@ -101,6 +130,21 @@ def commandline():
         logging.root.setLevel(logging.INFO)
     if args.debug:
         logging.root.setLevel(logging.DEBUG)
+    args_dict = vars(args)
+    # if command with remote_copy and remote_exec params
+    if 'remote_copy' in args_dict and 'remote_exec' in args_dict:
+        # if remote-exec and/or remote-copy args are None then user
+        # didn't specify command line options nor env vars:
+        # inspect multiplexing options for default cdist.REMOTE_COPY/EXEC
+        if args_dict['remote_copy'] is None or args_dict['remote_exec'] is None:
+            control_path_dir = tempfile.mkdtemp(prefix="cdist.control.path")
+            import atexit
+            atexit.register(lambda: shutil.rmtree(control_path_dir))
+            mux_opts = inspect_ssh_mux_opts(control_path_dir)
+            if args_dict['remote_exec'] is None:
+                args.remote_exec = cdist.REMOTE_EXEC + mux_opts
+            if args_dict['remote_copy'] is None:
+                args.remote_copy = cdist.REMOTE_COPY + mux_opts
```

Comment:
You mean on directory that Local class (cdist/exec/local.py) uses for base_path, right?
It uses path from args.out_path. If base_path is not specified then it creates temp directory.
We could do this outside of Local class.
When script/cdist parses args and in the case of "config" command if args.out_path is not specified then script/cdist creates args.out_path temp directory which Config later uses it for Local's base_path and cdist script also uses it for ssh mux socket.
In this way this temp directory is created in the one place and used for both socket and temp files.
What do you think?